File: LeetCode/011 Container With Most Water/Solution.java
public class Solution {
    public int maxArea(int[] height) {
        int width = height.length;
        int low = 0;
        int high = width - 1;
        int maxArea = 0;
        while (low<high){
            maxArea = Math.max(maxArea, (high-low)*Math.min(height[low],height[high]));
            if(height[low]<height[high]){
                low++;
            }
            else 
                high--;
        }
        return maxArea;
    }
}

File: LeetCode/011 Container With Most Water/Solution.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxArea, i, j = 0, 0, len(height) - 1

        while i < j:
            maxArea = max(maxArea, min(height[i], height[j]) * (j - i))
            if height[i] < height[j]:
                i +=1
            else:
                j -=1

        return maxArea

if __name__ == "__main__":
    height = [1, 2, 3, 4, 3, 2, 1, 5]
    result = Solution().maxArea(height)
    print(result)
